Dockerfile

```diff
@@ -0,0 +1,102 @@
+FROM ruby:3.2 AS builder
+
+RUN apt-get update && apt-get upgrade -y && apt-get install -y ca-certificates curl gnupg && \
+    curl -fsSL https://deb.nodesource.com/setup_21.x | bash - && \
+    apt-get install -y \
+    build-essential \
+    nodejs \
+    postgresql-client \
+    libpq-dev && \
+    apt-get clean
+
+# throw errors if Gemfile has been modified since Gemfile.lock
+RUN bundle config --global frozen 1
+
+WORKDIR /app
+
+# Copy package dependencies files only to ensure maximum cache hit
+COPY ./Gemfile /app/Gemfile
+COPY ./Gemfile.lock /app/Gemfile.lock
+
+RUN gem install bundler:$(grep -A 1 'BUNDLED WITH' Gemfile.lock | tail -n 1 | xargs) && \
+    bundle config --local without 'development test' && \
+    bundle install -j4 --retry 3 && \
+    # Remove unneeded gems
+    bundle clean --force && \
+    # Remove unneeded files from installed gems (cache, *.o, *.c)
+    rm -rf /usr/local/bundle/cache && \
+    find /usr/local/bundle/ -name "*.c" -delete && \
+    find /usr/local/bundle/ -name "*.o" -delete && \
+    find /usr/local/bundle/ -name ".git" -exec rm -rf {} + && \
+    find /usr/local/bundle/ -name ".github" -exec rm -rf {} + && \
+    find /usr/local/bundle/ -name "spec" -exec rm -rf {} +
+
+# copy the rest of files
+COPY ./app /app/app
+COPY ./bin /app/bin
+COPY ./config /app/config
+COPY ./db /app/db
+COPY ./lib /app/lib
+COPY ./public/*.* /app/public/
+COPY ./config.ru /app/config.ru
+COPY ./Rakefile /app/Rakefile
+
+# Compile assets with Webpacker or Sprockets
```

Comment:
We are not using Webpacker (in fact this library is deprecated), maybe we can remove references to it to avoid confussions? My idea is to move to `importmap-rails` or `jsbundling-rails` in the future.

Reply:
I've removed the comments, this is harmless anyway, just a safe guard